testing/helpers/test-helpers.bash

```diff
@@ -45,3 +54,104 @@ function find_unused_port() {
         fi
     done
 }
+
+# Converts the value of $COUCHBASE_SERVER_HOSTS into a file ready to feed to Prometheus file_sd.
+# Stores the name of the generated file in $PROMETHEUS_TARGETS_FILE.
+function _create_prometheus_targets_file() {
+    local tmp
+    tmp=$(mktemp -d)
+    echo "$COUCHBASE_SERVER_HOSTS" | jq -R -s '[split("\n")[:-1][] | sub(":8091"; ":9091")] | [{"targets": ., "labels": {"job":"couchbase-server","cluster":"smoke"}}]' > "$tmp/smoke.json"
+    export PROMETHEUS_TARGETS_FILE="$tmp/smoke.json"
+}
+
+# Starts a docker container named `cmos`, mounting $PROMETHEUS_TARGETS_FILE.
+# Exposes a variable $CMOS_HOST with the nginx host:port.
+# All parameters will be passed on to docker before the image.
+function _start_cmos() {
+    docker run --rm -d -p '8080' --name cmos "$@" "$CMOS_IMAGE"
```

Comment:
Are we ok with 8080 always?

Reply:
Now that you mention it, we should be using the random port like we do for k8s tests. Will change.